source/common/res/features/split-transaction-auto-adjust/main.js

```diff
@@ -0,0 +1,109 @@
+(function poll() {
+  // Waits until an external function gives us the all clear that we can run (at /shared/main.js)
+  if (typeof ynabToolKit !== 'undefined' && ynabToolKit.pageReady === true) {
+    ynabToolKit.splitTransactionAutoAdjust = (function () {
+      var addAnotherSplit;
+      var splitTransactionRow;
+      var isInitialized = false;
+      var addingAnotherSplit = false;
+      var deletingSplit = false;
+
+      function initialize() {
+        splitTransactionRow = $('.ynab-grid-add-rows');
+        addAnotherSplit = $('.ynab-grid-split-add-sub-transaction');
+
+        splitTransactionRow.on('keyup', '.currency-input .ember-text-field', onKeyPress);
+        splitTransactionRow.on('click', '.ynab-grid-sub-remove', onDeleteSplit);
+        addAnotherSplit.on('click', onAddAnotherSplit);
+      }
+
+      function onKeyPress() {
+        var element = this;
+        var currentInputClass = getCurrentInputClass();
+
+        if ($(element).parents(currentInputClass).length) {
+          autoFillNextRow(this);
+        }
+      }
+
+      function getCurrentInputClass() {
+        var firstRow = $('.ynab-grid-body-row', splitTransactionRow).first();
+        var outflowValue = parseFloat($('.ynab-grid-cell-outflow .ember-text-field', firstRow).val());
+        var inflowValue = parseFloat($('.ynab-grid-cell-inflow .ember-text-field', firstRow).val());
+        return outflowValue > 0 ? '.ynab-grid-cell-outflow' :
+               inflowValue > 0 ? '.ynab-grid-cell-inflow' : false;
+      }
+
+      function onAddAnotherSplit() {
+        addingAnotherSplit = true;
+      }
+
+      function onDeleteSplit() {
+        deletingSplit = true;
+      }
+
+      function autoFillNextRow(currentInputElement) {
+        var inputClass = getCurrentInputClass();
+        var total = parseFloat($(inputClass + ' .ember-text-field', splitTransactionRow.children().eq(0)).val());
+
+        if (inputClass && total) {
+          var currentRow = $(currentInputElement).parents('.ynab-grid-body-row');
+          var currentRowIndex = splitTransactionRow.children().index(currentRow);
+          var currentValue = parseFloat($(currentInputElement).val());
+
+          splitTransactionRow.children().each(function (index, splitRow) {
+            if (index === currentRowIndex) {
+              var nextRow = splitTransactionRow.children().eq(currentRowIndex + 1);
+              if (index === 0) {
+                $(inputClass + ' .ember-text-field', nextRow).val(total.toFixed(2));
```

Comment:
Ok, I'm pretty sure this one will cause issues in other locales. A good example of doing this that's been battle tested is in [this file](https://github.com/toolkit-for-ynab/toolkit-for-ynab/blob/master/source/common/res/features/budget-balance-to-zero/main.js).

So to parse a string and return a number: `ynab.unformat(amount)`
And to take a number and format it to a string, there's a formatting object hanging off of the `ynab` object, but it's honestly easier to just throw the raw number in there with `.val()`.